### PR TITLE
The Long Awaited Player Refactor

### PR DESCRIPTION
**Pull request by akrs**
I broke up the code into Actor, Player, and Controls classes.
- Controls has everything that's responsible for reading from inputs.
- Actor has general stuff that all 'moving' things should have
  - This should probably be expanded, and perhaps some of the abstract methods replaced with full methods.
- Player has player specific code.

I've also cleaned up variable names and such. Let me know what you think.

### CODE REVIEW

File: Game/Assets/FoF/Scripts/Player.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+			distToGround = GetComponent<Collider>().bounds.extents.y;
+			cameraTransform = transform.Find("ScientistSkeleton/Hips/Spine/Spine1/Neck/Head/Camera");   // Gotta love long identifiers
+			cameraRotX = cameraTransform.localEulerAngles.x;
+			controls = new Controls(this);
+		}
+		
+		// We shouldn't have to update here. Controls should move us around with its FixedUpdate()
+		void FixedUpdate ()
+		{
+			controls.FixedUpdate ();
```
Comment (akrs) [inline Game/Assets/FoF/Scripts/Player.cs:53]: Turns out, you can't create a new MonoActivity with new, so I ended up having to do this to get it to actually update.

**Comment by alexschneider** [comment]
I'd rather not keep the deprecated files around.

**Comment by ajohn104** [comment]
I'll be removing a lot of the unneeded files soon enough, I'd just rather remove it after we know it's totally useless. So if the files you're talking about are combat/camera/controls related, just keep 'em around for now.

File: Game/Assets/FoF/Scripts/Player.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
+﻿using UnityEngine;
+using System.Collections;
+using FoF;
+
+namespace FoF
```
Comment (ajohn104) [inline Game/Assets/FoF/Scripts/Player.cs]: We'd best update this to Coliseo now, rather than later down the line. Just to unify everything.

**Comment by ajohn104** [comment]
Well, other than my changes, I'm perfectly happy with this pull request.

**Pull request merged**